### PR TITLE
Fix entities/check/alert proxy

### PR DESCRIPTION
**Pull request by Jan-M**
Fixing this temporarily with these timeouts and new executor, not thinking about pool size right now.

This should tackle the threads getting stuck, blocking everything.

@jbellmann FYI

### CODE REVIEW

File: src/main/java/de/zalando/zmon/dataservice/proxies/checks/DefaultChecksService.java [hunk 1]
```diff
@@ -17,18 +21,31 @@ public DefaultChecksService(DataServiceConfigProperties config) {
...
 
+	public static HttpClient getHttpClient(int socketTimeout, int timeout, int maxConnections) {
+		RequestConfig config = RequestConfig.custom().setSocketTimeout(socketTimeout).setConnectTimeout(timeout).build();
+		return HttpClients.custom().setMaxConnPerRoute(maxConnections).setMaxConnTotal(maxConnections).setDefaultRequestConfig(config).build();
+	}
+
 	@Override
 	public String allActiveAlertDefinitions(String query) throws URISyntaxException, IOException {
+
+		final Executor executor = Executor.newInstance(getHttpClient(100, 5000, 1));
```
Comment (hjacobs) [inline src/main/java/de/zalando/zmon/dataservice/proxies/checks/DefaultChecksService.java]: 100ms socket timeout? that will probably not work, we also need to make these settings configurable (Spring properties class)

File: src/main/java/de/zalando/zmon/dataservice/proxies/entities/DefaultEntitiesService.java [hunk 2]
```diff
@@ -15,57 +17,60 @@
...
 
-	@Override
-	public String addEntities(String entities) throws URISyntaxException, IOException {
-		JsonNode node = customObjectMapper.readTree(entities);
-		if (node.has("infrastructure_account")) {
-			String id = node.get("infrastructure_account").textValue();
-			metrics.markEntity(id, 1);
-		}
+        final Executor executor = Executor.newInstance(getHttpClient(100, 5000, 1));
+        return executor.execute(Request.Get(uri)).returnContent().asString();
```
Comment (hjacobs) [inline src/main/java/de/zalando/zmon/dataservice/proxies/entities/DefaultEntitiesService.java]: there is also a lot of repeated code, executor and maybe execute could be done only once in a method (you already have `getHttpClient` as a helper method anyway)

**Comment by Jan-M** [comment]
I don't consider this the final solution, a lot of stuff need change, but this should def. fix / solve the current symptom of DS becoming unresponsive in case of downstream being unavailable/unresponsive.

**Comment by Jan-M** [comment]
:+1:

**Pull request merged**